clientd3d/d3drender.c

```diff
@@ -7131,14 +7149,16 @@ void D3DRenderObjectsDraw(d3d_render_pool_new *pPool, room_type *room,
 				pChunk->st1[3].s = D3DRENDER_CLIP_TO_SCREEN_X(topLeft.x, gScreenWidth) / gScreenWidth;
 				pChunk->st1[3].t = D3DRENDER_CLIP_TO_SCREEN_Y(topLeft.y, gScreenHeight) / gScreenHeight;
 
-				pChunk->st1[0].s -= (gFrame & 3) / 256.0f;
-				pChunk->st1[0].t -= (gFrame & 3) / 256.0f;
-				pChunk->st1[1].s -= (gFrame & 3) / 256.0f;
-				pChunk->st1[1].t += (gFrame & 3) / 256.0f;
-				pChunk->st1[2].s += (gFrame & 3) / 256.0f;
-				pChunk->st1[2].t += (gFrame & 3) / 256.0f;
-				pChunk->st1[3].s += (gFrame & 3) / 256.0f;
-				pChunk->st1[3].t -= (gFrame & 3) / 256.0f;
+				// animationFactor defines the intensity of the animation where lower values are more intense.

```

Comment:
This is still kind of confusing, and the amount of duplicated code is error-prone.  How about making an inline function near the top of this file, something like

```
// Good comment here describing what this is used for.
float animationIntensity(int frameNumber) 
{
  return (frameNumber & 3) / 256.0f; 
}
```

I think this will make it clearer what is going on and avoid cut-and-paste errors in the future.